mlflow/utils/validation.py

```diff
@@ -36,7 +36,7 @@
 
 _BAD_CHARACTERS_MESSAGE = (
     "Names may only contain alphanumerics, underscores (_), dashes (-), periods (.),"
-    " spaces ( ), and slashes (/)."
+    " spaces ( ), colon(:) and slashes (/)."
```

Comment:
On windows, can files and directory names contain `:`?

Reply:
good point. But Linux path does not allow "/" char too, why does it work? or is it not covered by test ?

Reply:
@piyushdaftary 
Good point . Could you add another test to log param / metric with key name including these good characters?

Reply:
Windows doesn't support `:`, so in windows, bad char list should still contain colon char.

Reply:
Good Point. I have refactored the code to handle `colon` for windows and non-windows platform differently. Please have a look. @WeichenXu123 @harupy